### PR TITLE
feat: accept GitHub URLs directly in CLI (lazy input)

### PR DESCRIPTION
**Pull request by hoanggiangle**
## Summary
- Paste GitHub links directly: `harness-pr-review https://github.com/owner/repo/pull/123`
- `autoreview --add-repo https://github.com/owner/repo --mode auto`
- Also accepts `owner/repo#123` and plain `owner/repo 123`
- New `src/repo_ref.py` with tests (6 tests)
- `doctor` unchanged; README updated with link examples
- Fixed pip-install package imports (src.* namespace) + colored PR comment sections

## Test Plan
- [x] pytest: 131 passed (6 repo_ref tests + 1 autoreview URL test + 1 bare-name fallback test)
- [x] Smoke: `harness-pr-review https://github.com/nexpeakcore/erp/pull/935 --no-post --skip-human --dry-run` → report generated
- [x] Verified pip install from git URL works (fresh venv)

### CODE REVIEW

**Comment by hoanggiangle** [comment]
## Harness PR Review — Verdict: <span style="background-color:#27ae60;color:#fff;padding:2px 10px;border-radius:10px;font-size:12px;font-weight:600">ACCURATE</span>

<details open><summary style="cursor:pointer;font-weight:600;font-size:14px;margin:8px 0">🟢 Claims (7)</summary>

<table style="border-collapse:collapse;width:100%"><thead><tr><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Claim</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Content</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Category</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Status</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Evidence</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Notes</th></tr></thead><tbody><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">C1</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">The CLI accepts a full GitHub PR URL as a positional argument, e.g. `harness-pr-review https://github.com/owner/repo/pull/123`.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">feature</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">PASS</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">src/run.py:175-176, src/repo_ref.py:23-28, tests/test_repo_ref.py:24-26</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">run.py routes strings containing /pull/ to parse_pr, which extracts owner/repo/PR from full GitHub PR URLs; manual fixture run with a URL succeeded.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">C2</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">`autoreview` accepts a GitHub repository URL through `--add-repo`, e.g. `autoreview --add-repo https://github.com/owner/repo --mode auto`.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">feature</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">PASS</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">src/autoreview.py:185-198, src/repo_ref.py:11-20, tests/test_autoreview.py:222-230</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">--add-repo parses GitHub repo URLs via parse_repo, writes normalized owner/repo key, and test_main_add_repo_url covers it.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">C3</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">The CLI also accepts `owner/repo#123` and plain `owner/repo 123` as repository references.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">feature</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">PASS</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">src/run.py:175-181, src/repo_ref.py:29-34, tests/test_run.py:27, tests/test_run.py:46, tests/test_repo_ref.py:29-30</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">owner/repo#N is parsed by the # branch and owner/repo N remains supported through the separate number argument; both were run successfully with fixtures.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">C4</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">A new `src/repo_ref.py` module was added for repository reference parsing, with 6 tests in `tests/test_repo_ref.py`.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">feature</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">PASS</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">src/repo_ref.py:1-35, tests/test_repo_ref.py:7-35</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">src/repo_ref.py exists with parse_repo/parse_pr; tests/test_repo_ref.py contains exactly six test functions.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">C5</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Pip-install package imports are fixed to use the `src.*` namespace.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">bugfix</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">PASS</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">pyproject.toml:15-17, pyproject.toml:23-25, src/run.py:12-16, src/autoreview.py:15-18, src/autoreview_config.py:100, src/claims.py:6, src/snapshot.py:6, src/synthesize.py:4</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">All internal src imports now use the src.* namespace and package discovery includes src*, fixing pip-installed package imports.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">C6</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">PR comment sections are colored.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">ux</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">PASS</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">src/synthesize.py:88-110, src/synthesize.py:129-153, src/synthesize.py:206-233, tests/test_synthesize.py:63-67, tests/test_synthesize.py:139-144</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">build_comment emits colored badges and colored status cells inside collapsible details sections; tests assert background-color and &lt;details&gt;.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">C7</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">README is updated with examples of GitHub link inputs.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">PASS</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">README.md:113, README.md:116</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">README Usage includes both harness-pr-review GitHub PR URL and autoreview --add-repo URL examples.</td></tr></tbody></table>

</details>
<details><summary style="cursor:pointer;font-weight:600;font-size:14px;margin:8px 0">📄 Docs vs reality (9)</summary>

<table style="border-collapse:collapse;width:100%"><thead><tr><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Doc</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Status</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Difference</th></tr></thead><tbody><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">README.md</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">STALE</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">GitHub-link examples are correct, but README still calls the dashboard read-only while server supports config management and Review now/re-review.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/index.html</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">MATCH</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Quickstart includes GitHub PR URL, owner/repo form, and current install/run commands; no contradiction with code.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/designs/2026-08-16-repo-config-design.md</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">MATCH</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">--add-repo URL acceptance, config format, and repo management APIs match implemented code.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/designs/2026-08-16-auto-review-design.md</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">STALE</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Still shows legacy list-form config and older log/status wording; current autoreview uses dict config and URL add-repo support.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/designs/2026-08-16-repo-open-prs-design.md</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">STALE</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Most open-PR/rounds behavior matches, but the documented &#x27;Bugs (based on N reviewed PRs)&#x27; label is not rendered by repo.html.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/designs/2026-08-16-review-status-log-design.md</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">STALE</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Status/log APIs match, but UI shows started_at rather than elapsed time and reloads after log stops, not every 15s while reviewing.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/guide/how-to-automate-pr-review-with-deepseek.html</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">STALE</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Guide describes the dashboard as read-only; current web UI includes config management and Review now/re-review actions.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/guide/verify-pr-descriptions-with-llm.html</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">MATCH</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Claim splitting, verdicts, evidence, and unverified-question flow are consistent with current pipeline behavior.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">docs/guide/detect-stale-docs-in-codebase.html</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#27ae60;font-weight:600">MATCH</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Doc statuses and human-confirmation flow are broadly consistent with current verify and synthesize behavior.</td></tr></tbody></table>

</details>
<details><summary style="cursor:pointer;font-weight:600;font-size:14px;margin:8px 0">🎯 Requirement impact (8)</summary>

<table style="border-collapse:collapse;width:100%"><thead><tr><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Requirement</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Impact</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Detail</th></tr></thead><tbody><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">CLI PR input parsing</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">CHANGED</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">harness-pr-review now accepts full GitHub PR URLs and owner/repo#N in addition to existing owner/repo N.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Auto-review repo configuration</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">CHANGED</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">--add-repo now accepts full GitHub repository URLs and stores normalized owner/repo keys; bare-name fallback remains.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Pip-installed package imports</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">CHANGED</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Internal imports and console entry points use the src.* namespace so installed commands resolve modules correctly.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">PR comment rendering</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">CHANGED</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">PR comments now use colored HTML badges/cells inside collapsible sections instead of plain markdown summaries.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Web dashboard interactivity</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">CHANGED</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Dashboard now includes config management, repo pages without prior sessions, and review-trigger endpoints beyond read-only display.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Doctor command</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#6b7280;font-weight:600">UNAFFECTED</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Doctor behavior is unchanged and still checks Python, gh, API key, SDK, and autoreview.yml.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Legacy CLI workflows</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#6b7280;font-weight:600">UNAFFECTED</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Existing owner/repo N and owner/repo#N invocations continue to work.</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Documentation consistency</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top"><span style="color:#b9770e;font-weight:600">RISK</span></td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">README and how-to guide still describe a read-only dashboard, and some design docs have minor outdated details.</td></tr></tbody></table>

</details>
<details><summary style="cursor:pointer;font-weight:600;font-size:14px;margin:8px 0">💬 Review threads</summary>

<table style="border-collapse:collapse;width:100%"><thead><tr><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Comment</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Status</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Note</th></tr></thead><tbody></tbody></table>

</details>
<details><summary style="cursor:pointer;font-weight:600;font-size:14px;margin:8px 0">✅ Confirm log (2)</summary>

<table style="border-collapse:collapse;width:100%"><thead><tr><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Question</th><th style="padding:6px 10px;border:1px solid #e3e8ee;background:#f6f8fa;text-align:left;font-size:12px">Answer</th></tr></thead><tbody><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Was pip install from a git URL actually exercised? Only static import verification possible offline.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">SKIPPED</td></tr><tr><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">Do GitHub PR comments render inline color styles as expected? Not verified in live GitHub.</td><td style="padding:6px 10px;border:1px solid #e3e8ee;font-size:13px;vertical-align:top">SKIPPED</td></tr></tbody></table>

</details>

Full report: local `sessions/nexpeakcore/deepseek-harness-pr-review/pr-3/report.md`

File: src/run.py [hunk 1]
```diff
@@ -9,11 +9,11 @@
 import sys
```
Comment (nv-minh) [inline src/run.py:9]: **Test coverage — WEAK**

`src/run.py:main (positional-arg dispatch logic)`: test_main_owner_repo_hash_parsing only checks exit code 0 and report existence, not that parse_pr/parse_repo were actually invoked or that the wrong branch wasn't silently taken; no test drives main() with a full '/pull/123' URL end-to-end.

Uncovered:
- main() invoked with a full GitHub PR URL as args[0] (only repo_ref unit tests cover parsing, not the CLI dispatch branch in run.py:175-176) — add to `tests/test_run.py`
- main() invoked with both a '#'-style ref AND a stray second positional number, to confirm precedence — add to `tests/test_run.py`

File: src/synthesize.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 """Phase 5: synthesize the report + post an English comment on the PR."""
```
Comment (nv-minh) [inline src/synthesize.py:1]: **Test coverage — WEAK**

`src/synthesize.py:build_comment (colored sections)`: Tests check for the presence of 'background-color' and '<details' substrings but never assert which color maps to which status, or that FAIL/WRONG/BROKEN get the red palette vs PASS getting green.

Uncovered:
- STATUS_COLORS lookup miss (status value not in the dict) falling back silently to no color — add to `tests/test_synthesize.py:test_build_comment_html_sections`

**Comment by nv-minh** [comment]
## PR Sentinel — Verdict: ACCURATE

**Gate: WARNING** · Verification score 100% · Business risk none

- 1 core doc(s) out of sync: docs/guide/how-to-automate-pr-review-with-deepseek.html

<details>

<summary>Findings that could not be anchored to the diff (1)</summary>

**`docs/guide/how-to-automate-pr-review-with-deepseek.html`**

**Docs out of sync** — src/run.py's main() now accepts a GitHub PR URL (containing '/pull/') or 'owner/repo#123' shorthand in addition to the separate 'owner/repo 123' args (see parse_pr/parse_repo branching on '#' or '/pull/' in args.pr), but the doc still only shows the old two-argument form.

```suggestion
<h2>Step 2 — Review one PR</h2>
<pre><code>python -m src.run owner/repo 123
python -m src.run owner/repo#123
python -m src.run https://github.com/owner/repo/pull/123</code></pre>
```

</details>

<details open>

<summary>Full report</summary>

# Review PR #3 — feat: accept GitHub URLs directly in CLI (lazy input)

- Author: hoanggiangle | Base: main → Head: feat/lazy-repo-link
- Files changed: 38 | Commits: 13
## Verdict: ACCURATE

- Gate: **WARNING** | Verification score: 100% | Business risk: none
  - 1 core doc(s) out of sync: docs/guide/how-to-automate-pr-review-with-deepseek.html

- Requirement: none available — Jira is not configured (JIRA_BASE_URL, JIRA_EMAIL, JIRA_API_TOKEN)

## Claims

| Claim | Content | Status | Evidence | Notes |
|---|---|---|---|---|
| C1 | New file src/repo_ref.py created with functionality to parse GitHub URLs and repo references | Matches | src/repo_ref.py:1-36, tests/test_repo_ref.py:1-36 | repo_ref.py implements parse_repo/parse_pr with regex-based URL and shorthand parsing; test file exists with 6 tests. |
| C2 | CLI accepts full GitHub pull request URLs in format: https://github.com/owner/repo/pull/123 | Matches | src/repo_ref.py:4-6, src/repo_ref.py:26-28, src/run.py:172-181 | run.py main() detects '/pull/' in the CLI arg and calls parse_pr, which matches _REPO_URL with the pr group. |
| C3 | CLI accepts shorthand owner/repo#PR_number format (e.g., owner/repo#123) | Matches | src/repo_ref.py:29-34, src/run.py:175-176 | parse_pr splits on '#' and validates the numeric suffix; run.py routes to parse_pr when '#' is present. |
| C4 | CLI accepts plain owner/repo with separate PR number (e.g., owner/repo 123) | Matches | src/run.py:177-179, tests/test_run.py:19-31 | When a second positional number is supplied, run.py calls parse_repo(args.pr) and uses args.number as the PR id; covered by test_main_fixtures_mode. |
| C5 | Test coverage includes 6 tests for repo_ref module in tests/test_repo_ref.py | Matches | tests/test_repo_ref.py:7-36 | Exactly 6 test functions: test_parse_repo_url_full, test_parse_repo_no_scheme, test_parse_repo_invalid, test_parse_pr_url, test_parse_pr_shorthand, test_parse_pr_invalid_number. |
| C6 | autoreview command accepts --add-repo parameter with GitHub URL format | Matches | src/autoreview.py:185-199, tests/test_autoreview.py:222-231 | --add-repo runs parse_repo() on the raw string, falls back to bare-name/org handling on ValueError; test_main_add_repo_url exercises the full GitHub URL path. |
| C7 | Fixed package imports to use src.* namespace for pip-installed packages | Matches | src/autoreview.py:15-18, src/autoreview_config.py:100, src/run.py:12-16, src/synthesize.py:4, src/snapshot.py:6, src/claims.py:6, web/server.py:12-15, web/metrics.py:7 | Every internal import across src/ and web/ uses the 'src.*' namespace consistently (needed for pip-installed console_scripts entry points where 'src' is a top-level package). |
| C8 | Added colored sections to PR comment output | Matches | src/synthesize.py:88-153, tests/test_synthesize.py:58-66, tests/test_synthesize.py:139-143 | build_comment now renders HTML tables with inline background-color/status badges inside collapsible <details> sections; tests assert 'background-color' and '<details' are present. |
| C9 | README updated with GitHub URL link examples | Matches | README.md:104-119 | Usage section now shows 'harness-pr-review https://github.com/owner/repo/pull/123' and 'autoreview --add-repo https://github.com/owner/repo --mode auto'. |
| C10 | doctor command implementation remains unchanged | Matches | src/run.py:19-72, tests/test_run.py:141-173 | _doctor() body reads as a standard, self-contained readiness check with no repo_ref/URL-parsing changes; doctor tests are unchanged in behavior. Cannot diff against the pre-PR version directly, so tre… |

## Docs vs reality

| Doc | Status | Difference |
|---|---|---|
| README.md | MATCH | Usage section correctly documents the new URL / '#' / plain forms exactly as implemented in src/repo_ref.py and src/run.py. |
| docs/index.html | MATCH | Quickstart block already shows 'harness-pr-review https://github.com/owner/repo/pull/123' alongside the classic 'owner/repo 123' form, matching src/run.py:172-181. |
| docs/designs/2026-08-16-repo-config-design.md | MATCH | Design doc's --add-repo bullet now explicitly notes 'URLs like https://github.com/owner/repo accepted', matching src/autoreview.py:185-199. |
| docs/guide/how-to-automate-pr-review-with-deepseek.html | STALE | Step 2 still only shows 'python -m src.run owner/repo 123' and doesn't mention the new GitHub-URL / '#'-shorthand input this PR adds, even though the file was touched (+1/-1) in this PR. |

## Requirement impact

| Requirement | Impact | Area | Detail |
|---|---|---|---|
| Accept GitHub PR URLs / shorthand refs as CLI input | CHANGED | other | CLI now parses full GitHub URLs, owner/repo#n, and owner/repo + n via src/repo_ref.py before dispatching the existing 5-phase pipeline; no change to downstream phases. |
| autoreview --add-repo accepts GitHub URLs | CHANGED | infra | --add-repo now runs the value through parse_repo() before falling back to bare-name+org handling. |
| pip-installed package imports use src.* namespace | CHANGED | infra | All internal imports were normalized to 'from src.X import ...', required for the console_scripts entry points (harness-pr-review = src.run:main) to work when installed via pip; pyproject.toml already… |
| Colored PR comment sections | CHANGED | other | build_comment() now emits HTML tables with colored status badges inside <details> sections instead of plain markdown; report.md format (build_report) is unaffected. |
| inferred from code: autoreview.yml is the tracked production config for the poller | RISK | infra | The committed autoreview.yml gained 'admin-web: auto', 'sample-org/sample-api: auto' and 'sample-app2: auto' entries — these names are identical to the fixtures used in tests/test_autoreview.py (test_… |

## Callers outside the diff

| Symbol | Defined at | Callers | Risk | Notes |
|---|---|---|---|---|
| src.repo_ref.parse_repo / parse_pr | src/repo_ref.py:11-35 | src/run.py:178,181 (in diff), src/autoreview.py:190 (in diff) | SAFE | Only callers are within this PR's diff (run.py, autoreview.py); no external callers found repo-wide via grep. |
| src.synthesize.build_comment / build_report | src/synthesize.py:36,156 | src/run.py:241,246 (in diff), tests/test_synthesize.py (in diff) | SAFE | Signature (positional args) unchanged; only the internal HTML/body construction changed. web/metrics.py only imports _overall_verdict, whose signature is untouched. |
| src.synthesize._overall_verdict | src/synthesize.py:25 | web/metrics.py:100 (in diff, +2/-2) | SAFE | Function signature and return values (ACCURATE/PARTIAL/MISLEADING/NO CLAIMS) unchanged; metrics.py still calls it the same way. |
| src.run.main | src/run.py:140-260 | web/server.py:346 run_main(args) (in diff), src/autoreview.py:118 _dispatch (in diff), scripts/autoreview-once.sh (calls src.autoreview, not src.run, directly) | SAFE | All in-repo callers pass '<owner>/<repo> <number> ...' which is still accepted via the args.number branch (src/run.py:177-179); scripts/autoreview-once.sh never calls src.run with positional args at a… |

## Contract changes

- No API, schema, type or proto contract touched.

## Cross-PR collisions

- No collision found with the open pull requests listed below.

## Parallel open pull requests

- no other open pull request changes the same files or a sensitive module this PR touches

- Scanned 0 open pull request(s).

## Test integrity

| Target | Assertions | Uncovered edge cases | Notes |
|---|---|---|---|
| src/repo_ref.py:parse_repo, parse_pr | STRONG | URL with trailing slash / no repo segment, e.g. 'owner/repo/' or 'https://github.com/owner/' → tests/test_repo_ref.py:test_parse_repo_invalid<br>parse_pr on a plain 'owner/repo' with no '#' and no URL pr segment (should raise, currently unverified by a dedicated te… → tests/test_repo_ref.py<br>Mixed input: URL with '/pull/N' AND a trailing '#M' — which one wins is untested → tests/test_repo_ref.py | Tests assert exact (owner, repo) / (owner, repo, pr) tuples for URL, no-scheme, shorthand and invalid inputs, including the ValueError message via pytest.raises(match=...). |
| src/run.py:main (positional-arg dispatch logic) | WEAK | main() invoked with a full GitHub PR URL as args[0] (only repo_ref unit tests cover parsing, not the CLI dispatch branch… → tests/test_run.py<br>main() invoked with both a '#'-style ref AND a stray second positional number, to confirm precedence → tests/test_run.py | test_main_owner_repo_hash_parsing only checks exit code 0 and report existence, not that parse_pr/parse_repo were actually invoked or that the wrong branch wasn't silently taken; no test drives main()… |
| src/synthesize.py:build_comment (colored sections) | WEAK | STATUS_COLORS lookup miss (status value not in the dict) falling back silently to no color → tests/test_synthesize.py:test_build_comment_html_sections | Tests check for the presence of 'background-color' and '<details' substrings but never assert which color maps to which status, or that FAIL/WRONG/BROKEN get the red palette vs PASS getting green. |

## Review threads

- (none)

## Excluded from review context

| File | Reason | Dropped |
|---|---|---|
| docs/plans/2026-08-16-auto-review.md | patch truncated to 400 of 557 lines | patch only |
| docs/plans/2026-08-16-repo-config.md | patch truncated to 400 of 930 lines | patch only |
| docs/plans/2026-08-16-repo-open-prs.md | patch truncated to 400 of 523 lines | patch only |
| docs/plans/2026-08-16-web-dashboard.md | patch truncated to 400 of 915 lines | patch only |

## Confirmation log

- **Were 'admin-web', 'sample-org/sample-api', 'sample-app2' in autoreview.yml intentionally committed, or leaked from local test runs?** → SKIPPED
- **Was src/run.py's _doctor() function actually untouched by this PR, or only reviewed post-hoc (no prior version available to diff)?** → SKIPPED
- **Should docs/guide/how-to-automate-pr-review-with-deepseek.html be updated to show the new URL/shorthand input forms?** → SKIPPED

_Review cost: $2.6987_


</details>
